src/gpio.rs

```diff
@@ -502,6 +518,43 @@ macro_rules! gpio {
                         unsafe { (*$GPIOX::ptr()).idr.read().bits() & (1 << $i) == 0 }
                     }
                 }
+
+                impl<MODE> ExtiPin for $PXi<Input<MODE>> {
+                    /// generate interrupt on rising edge, falling edge or both
+                    fn trigger_on_edge(&mut self, exti: &mut EXTI, edge: Edge) {
+                        match edge {
+                            Edge::RISING => {
+                                exti.rtsr.write(|w| { w.$tri().set_bit()   });
+                                exti.ftsr.write(|w| { w.$tri().clear_bit() });
+                            },
+                            Edge::FALLING => {
+                                exti.ftsr.write(|w| { w.$tri().set_bit()   });
+                                exti.rtsr.write(|w| { w.$tri().clear_bit() });
+                            },
+                            Edge::RISING_FALLING => {
+                                exti.rtsr.write(|w| { w.$tri().set_bit() });
+                                exti.ftsr.write(|w| { w.$tri().set_bit() });
+                            }
+                        }
+                    }
+
+                    /// Enable external interrupts from this pin.
+                    /// Note: This configures EXTI line $i to trigger from this pin
+                    fn enable_interrupt(&mut self, exti: &mut EXTI, syscfg: &mut SYSCFG) {
+                        syscfg.$exticri.modify(|r, w| unsafe {
+                            let mut exticr = r.bits();
+                            exticr = (exticr & $extigpionr) | $extigpionr;
```

Comment:
Unless I'm missing something, this line looks pretty wrong to me. It is actually equivalent to simply writing `exticr = $extigpionr`.

I guess you would need something like this instead:
```
let offset = 4 * ($extigpionr % 4);
exticr = (exticr & !(0xF << offset)) | ($extigpionr << offset);
```

Reply:
Thanks for catching this. Indeed the code is wrong here.